### ISSUE TITLE
GraalVM Enterprise version?

### ISSUE THREAD

**Issue by matus-gazo** [paketo-buildpacks/graalvm#210]
Hi there,

I want to use the Enterprise version with the buildpack to use PGO a G1 GC, but it seems there's no way to do that with the buildpack, which pretty much ruins the build pipeline I've set up for our product. Is there any way or any plans to be able to do this?

**Comment by dmikusa-pivotal** [comment]
You are correct that we do not currently support the enterprise version of GraalVM.

I think the biggest challenge is with the way Oracle distributes it. From what I can tell on their site, it's behind a EULA and user login. The buildpack needs to be able to download the software it installs, so while not impossible, that makes things more difficult.

I wouldn't be opposed to adding support for it, but we'll have to figure out a way for the buildpack to get the software. That would likely require you, as a user, to download the software and host it somewhere internally/private that the buildpack can access. This way the buildpack doesn't need to worry about the Oracle site. Would this work for you?

Or are you aware if Oracle provides an API through which we could download the binaries or some alternative channel for downloads?

Thanks

**Comment by dmikusa** [comment]
It looks like GraalVM has a way to download EE versions, we should be able to do the same thing with the buildpack. See https://medium.com/graalvm/one-line-graalvm-jdk-downloads-91ee8d4e6ffd.

There's a bash script to download, but I think we can just reimplement what it's doing in Go. There's a token it's grabbing and a file it's downloading. Shouldn't be too hard.

**Comment by lgklein** [comment]
Hello @dmikusa, we have any ETA for this to be incorporate? This is a really welcome feature for company's builds...

**Comment by dmikusa** [comment]
@lgklein I can tell you that this is not something we've discussed implementing as a team, but that we'd be happy to help if anyone wants to contribute a PR for this. Just let me know, and I can provide an outline of what needs to be done.

**Comment by danthe1st** [comment]
Oracle has introduced a [new license](https://blogs.oracle.com/java/post/graalvm-free-license) changing the name of GraalVM Enterprise Edition to Oracle GraalVM and making it available for free (depending on the version).

**Comment by dmikusa** [comment]
Oh, interesting. Thanks for sharing!

**Comment by sdeleuze** [comment]
Could we maybe rename this issue to something like "Oracle GraalVM support" or something approaching, to match the new naming?

**Comment by dmikusa** [comment]
If you want to be involved in this conversation, we're discussing and planning this on the following RFC: https://github.com/paketo-buildpacks/rfcs/pull/294

Comments and feedback is appreciated. Thanks
